### DISCUSSION TITLE
style-loader: what's the point of enabling `esModule`?

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#18093] [Q&A] [answered]
As title. If those CSS scripts have been inserted into the DOM by adding `<style>` tag(s), would this mean that there is no more `import` so there is nothing to transform by ES Module syntax?

**Comment by alexander-akait** [comment, marked as answer]
When you set `esModule: true`, style-loader generates `import` and `export` syntax, it doesn't really affect anything if you don't have others loaders between style-loader and css-loader

**Reply by nyngwang** [reply]
I probably figured it out myself. In short, before style-loader does its things it still needs to have some `import`,`export` syntax there.
